test/json/json_addition_test.rb

```diff
@@ -152,7 +152,7 @@ def test_core
 
   def test_deprecated_load_create_additions
     pattern = /json_addition_test\.rb.*use JSON\.unsafe_load/
-    if RUBY_ENGINE == 'truffleruby'
+    if RUBY_ENGINE == 'truffleruby' || File.basename($0) != 'json_addition_test.rb'
```

Comment:
I don't think that is correct, because even in this repo, when we run `bundle exec rake` I don't think `$0` is `json_addtion_test.rb`.

Can you show me a failure from `ruby/ruby`? The PR you pointed seem to have passed fine.